Source/JavaScriptCore/runtime/DateConstructor.cpp

```diff
@@ -71,6 +71,37 @@ static double millisecondsFromComponents(JSGlobalObject* globalObject, const Arg
     VM& vm = globalObject->vm();
     auto scope = DECLARE_THROW_SCOPE(vm);
 
+    auto toIntegerOrInfinity = [](double d) {
+        return trunc(std::isnan(d) ? 0.0 : d + 0.0);
```

Comment:
Seems a little strange to write `trunc` and `std::isnan`. How about `std::trunc`?